client/packages/system/src/Stock/Components/InventoryAdjustment/InventoryAdjustmentForm.tsx

```diff
@@ -0,0 +1,201 @@
+import React, { FC } from 'react';
+import {
+  TextWithLabelRow,
+  InputWithLabelRow,
+  InputWithLabelRowProps,
+  useTranslation,
+  Box,
+  NumericTextInput,
+  DialogButton,
+  useNotification,
+  AdjustmentDirectionInput,
+} from '@openmsupply-client/common';
+import { StockLineRowFragment, useInventoryAdjustment } from '../../api';
+import {
+  Adjustment,
+  InventoryAdjustmentReasonSearchInput,
+  usePackVariant,
+} from '../../..';
+import { InventoryAdjustmentDirectionInput } from './InventoryAdjustmentDirectionSearchInput';
+
+// TODO... use direction input in stocktake?
+const tempAsAdjustment = (direction: AdjustmentDirectionInput | null) => {
+  switch (direction) {
+    case AdjustmentDirectionInput.Addition:
+      return Adjustment.Addition;
+    case AdjustmentDirectionInput.Reduction:
+      return Adjustment.Reduction;
+    default:
+      return Adjustment.None;
+  }
+};
+
+interface InventoryAdjustmentFormProps {
+  stockLine: StockLineRowFragment;
+  onUpdate: (patch: Partial<StockLineRowFragment>) => void;
+}
+export const InventoryAdjustmentForm: FC<InventoryAdjustmentFormProps> = ({
+  stockLine,
+}) => {
+  const t = useTranslation('inventory');
+  const { success } = useNotification();
+
+  const { draft, setDraft, create } = useInventoryAdjustment(stockLine);
+
+  const { asPackVariant } = usePackVariant(
+    stockLine.itemId,
+    stockLine.item.unitName ?? null
+  );
+  const packUnit = asPackVariant(stockLine.packSize);
+
+  const saveDisabled = !draft.direction || draft.adjustBy === 0;
+
+  const save = async () => {
+    // TODO: handle error if no reason selected when reasons required
```

Comment:
assuming we'll do similar logic here to stocktake, backend will error if reasons are defined and one should be set here?